azure-pipelines.yml

```diff
@@ -224,6 +270,7 @@ stages:
       dependsOn:
         - Build
```

Comment:
do we still need the build step here? I think that the deploy step makes a build itself (so probably we should remove this one)

Reply:
The build step is meant ton check if changes of a PR break the build. What we can do is to run them with `--noEmit` to save some processing, maybe.

Anyway, by introducing the Release stage, the Build and Deploy stages actually compile different source codes, so we must keep them separated (unless we do some dirty patching on built code).

Reply:
What I meant is that we already have a build step here _into_ the deploy stage: https://github.com/pagopa/io-functions-admin/blob/master/azure-templates/deploy-steps.yml#L56 
So why do we want to make another build _before_ this one? (we build twice currently)
If the step inside deploy stage fails the deploy task stops anyway, isn't it?

Reply:
The deploy stage happens after the release is created, hence if it fails we have a _broken_ release to be manually deleted.

Test and build stages are there because we want some static checks before committing into a new release. I see that this leads to building twice, still I think it's the only way to cope with all the scenarios:
* broken build must not be released
* the code with the updated version must be deployed

Reply:
what if we move the release part after a successful build in the deploy stage?

Reply:
The artefact we deploy would not have the version bumped. Apart from not referencing the correct commit in history, we would face the following:
* for services, we would not expose the correct version on `/info` endpoint
* for packages, we would ship the wrong version (or we would even be unable to deploy)

Reply:
I think that deploy step must be changed to deploy the released version on github.